### PR TITLE
Port to Firefox

### PR DESCRIPTION
**Pull request by rasendubi**
Problem: I'm trying Firefox Quantum now and Heutagogy plugin does not work there.

Solution: port it. (Mostly fix issues with `react-chrome-redux`.)

### CODE REVIEW

File: app/actions/article.js [hunk 1]
```diff
@@ -29,9 +29,12 @@ export const REMOVE_ARTICLE_SUCCESS = 'REMOVE_ARTICLE_SUCCESS';
 export const REMOVE_ARTICLE_FAILURE = 'REMOVE_ARTICLE_FAILURE';
 
 
+const toJS = (x) => x.toJS ? x.toJS() : x;
```
Comment (drets) [inline app/actions/article.js]: I would avoid the method name shadowing and rename it to: `convertToNativeJavascript`

File: app/containers/App.js [hunk 2]
```diff
@@ -61,28 +62,30 @@ class App extends Component {
...
+        <SaveControl
+        article={this.props.article}
+        currentTab={this.props.currentTab}
+        fetchArticleState={this.props.fetchArticleState}
+        rememberArticle={this.props.rememberArticle}
+        rememberArticleState={this.props.rememberArticleState}
+        removeArticle={this.props.removeArticle}
+        removeArticleState={this.props.removeArticleState}
+        updateArticle={this.props.updateArticle}
+        updateArticleState={this.props.updateArticleState}
```
Comment (drets) [inline app/containers/App.js]: I would try props destructuring like so: `<SaveControl {...this.props} />`.

File: chrome/extension/background.js [hunk 3]
```diff
@@ -65,6 +65,9 @@ chrome.storage.local.get('state', (obj) => {
   const w = watch(store.getState);
 
   store.subscribe(w((newVal, oldVal) => {
+    newVal = Immutable.fromJS(newVal);
+    oldVal = Immutable.fromJS(oldVal);
```
Comment (drets) [inline chrome/extension/background.js]: Appending `Immutable.fromJS` everywhere looks like a hack, could we avoid it :thinking:

File: chrome/views/popup.pug [hunk 4]
```diff
@@ -5,7 +5,7 @@ html
     meta(charset='UTF-8')
     title Heutagogy
     style.
-      body { width: 275px; height: 45px; }
+      body { width: 275px; }
```
Comment (drets) [inline chrome/views/popup.pug:8]: Why removing height?

File: app/containers/App.js (hunk 2 above)
Comment (rasendubi) [reply app/containers/App.js]: `this.props.user` is not passed in.

File: chrome/extension/background.js (hunk 3 above)
Comment (rasendubi) [reply chrome/extension/background.js]: We could drop immutable.js altogether, so no conversion is needed. Alternatively, we could patch store to handle that conversion implicitly. (I would try avoiding both.)

File: chrome/views/popup.pug (hunk 4 above)
Comment (rasendubi) [reply chrome/views/popup.pug:8]: 45px is about a height of a single toggle on my screen, so I only see a single input field at a time. (This is even more inconvenient when we draw a login screen.) 

By removing fixed height here we let the form control its own height as needed.

File: app/containers/App.js (hunk 2 above)
Comment (drets) [reply app/containers/App.js]: I would pass the `user` then.

**Pull request merged**